erpnext/hr/doctype/holiday/holiday.json

```diff
@@ -44,17 +45,24 @@
   {
    "fieldname": "section_break_4",
    "fieldtype": "Section Break"
+  },
+  {
+   "default": "0",
+   "fieldname": "public",
+   "fieldtype": "Check",
+   "label": "Public"
```

Comment:
1. I have put it that way to differentiate company holidays from public holidays. 
2. Even if we check both checkbox together it still makes sense because there could be a public holiday on a weekend

That is the reason I have not suggested a select box. The extra check box provides better reporting as well